File: src/main/java/exnihilocreatio/registries/types/FluidFluidBlock.java
package exnihilocreatio.registries.types;

import exnihilocreatio.util.BlockInfo;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;

@AllArgsConstructor
@EqualsAndHashCode
public class FluidFluidBlock {

    @Getter
    private String fluidInBarrel;

    @Getter
    private String fluidOnTop;

    @Getter
    private BlockInfo result;
}

File: src/main/java/exnihilocreatio/registries/types/FluidFluidBlock.kt
package exnihilocreatio.registries.types

import exnihilocreatio.util.BlockInfo

// TODO: MOVE TO API PACKAGE ON BREAKING VERSION CHANGE
data class FluidFluidBlock(
        val fluidInBarrel: String? = null,
        val fluidOnTop: String? = null,
        val result: BlockInfo? = null
)